src/Libraries/RevitNodes/Elements/ElementType.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Autodesk.DesignScript.Runtime;
+using Autodesk.Revit.DB;
+using DSCore.IO;
+using DynamoServices;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// A Revit ElementType
+    /// </summary>
+    public class ElementType : Element
+    {
+        #region Internal properties
+
+        /// <summary>
+        /// An internal reference to the ElementType
+        /// </summary>
+        internal Autodesk.Revit.DB.ElementType InternalElementType
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalElementType; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Private constructor for the Element
+        /// </summary>
+        /// <param name="elementType"></param>
+        private protected ElementType(Autodesk.Revit.DB.ElementType elementType)
+        {
+            SafeInit(() => InitElementType(elementType));
+        }
+
+        /// <summary>
+        /// Initialize a ElementType element
+        /// </summary>
+        /// <param name="elementType"></param>
+        private void InitElementType(Autodesk.Revit.DB.ElementType elementType)
```

Comment:
since this method only calls the `InternalSetElementType` method, we could remove this and call that directly. unless i'm missing a reason this exists ?

or move the code from `InternalSetElementType` here and delete that ?